src/odemis/driver/piezomotor.py

```diff
@@ -0,0 +1,733 @@
+        #  and then do a second search at slow speed, to locate the index more precisely. Also, if the index is
+        #  somewhere in the middle, you'll need to handle the fact you might have picked the wrong direction
+        #  (in which case, after while you'll bump into the limit switch) and in such case, continue by going in
+        #  the other direction.
+
+    def _doMoveAbs(self, pos):
+        self._check_hw_error()
+        targets = {}
+        for axis, val in pos.items():
+            if self._closed_loop[axis]:
+                encoder_cnts = round(val / ENCODER_STEPSIZE)
+                speed_usteps = round(self._speed / self._wfm_stepsizes[axis])  # wfm steps / second
+                self._sendCommand(b'X%dT%d,%d' % (self._axis_map[axis], encoder_cnts, speed_usteps))
+                targets[axis] = encoder_cnts
+            else:
+                self._updatePosition()
+                target = val - self.position.value[axis]
+                wfm_steps = int(target / self._wfm_stepsizes[axis])  # number of waveform steps
+                usteps = int((target % self._wfm_stepsizes[axis]) * USTEPS_PER_WFM)   # number of µsteps
+                self._sendCommand(b'X%dJ%d,%d,%d' % (self._axis_map[axis], wfm_steps, usteps, speed_usteps))
+        self._waitEndMotion(targets)
+        self._updatePosition()
+
+    def _doMoveRel(self, shift):
+        self._check_hw_error()
+        targets = {}
+        self._updatePosition()
+        for axis, val in shift.items():
+            speed_usteps = round(self._speed / self._wfm_stepsizes[axis] * USTEPS_PER_WFM)  # steps / second
+            if self._closed_loop[axis]:
+                encoder_cnts = round(val / ENCODER_STEPSIZE)
+                # There are two possibilities: move relative to current position (XC) and move relative to
+                # target position (XR). We are moving relative to the current position (might be more intuitive
+                # if something goes wrong and we're stuck in the wrong position).
+                self._sendCommand(b'X%dC%d,%d' % (self._axis_map[axis], encoder_cnts, speed_usteps))
+                targets[axis] = self.position.value[axis] / ENCODER_STEPSIZE + encoder_cnts
+            else:
+                wfm_steps = int(val / self._wfm_stepsizes[axis])  # number of waveform steps
+                usteps = int((val % self._wfm_stepsizes[axis]) * USTEPS_PER_WFM)   # number of µsteps
+                self._sendCommand(b'X%dJ%d,%d,%d' % (self._axis_map[axis], wfm_steps, usteps, speed_usteps))
+                targets[axis] = None
+        self._waitEndMotion(targets)
+        self._updatePosition()
+
+    def _waitEndMotion(self, targets):
+        """
+        Wait until move is done
+        :arg targets (dict: str --> int): target (for closed-loop), None for open loop
+        """
+        # Expected time for move
+        move_length = sum([abs(self.position[ax] - targets[ax]) for ax in targets.keys()])
```

Comment:
You can drop the `[` as the sum accepts directly an iterator. You could also use `.items()` instead of `.keys()` to avoid the duplication of `targets`.
BTW, I guess that all the axes move simultaneously? In that case, it's not "sum" but "max" that you need to use.